README.md

```diff
@@ -72,3 +72,82 @@ meaningful. The blog post itself then can contain a longer title specified in th
+is basically just a wrapper for embedding the GitHub discussions into another page.
+
+By default discussions are enabled below all blog posts. To disable discussion under a specific
+blog post set `discussion: false` in the blog post front matter.
+
+For blog posts discussions it was created a special
+[Blog](https://github.com/agama-project/agama-project.github.io/discussions/categories/blog)
+discussion category to easily distinguish the blog posts discussions from others.
+
+### Permissions
+
+The new blog posts discussions can be created by Agama developers, the other users can create them
+only via Giscus. All interactions from the blog post page are done through the Giscus application.
+That means the users need to authorize the Giscus application to post on their behalf.
+
+If you do not want to authorize the Giscus you can comment directly at the respective existing
+GitHub discussion without using the Giscus proxy.
+
+### Mapping
+
+There two ways how to find the respective GitHub discussion for a blog post. Each approach has
+different pros and cons so the Agama blog supports both.
+
+#### Search blog post URL path
+
+The respective discussion at GitHub is found by searching the URL path in the discussion title.
+E.g. for the `https://agama-project.github.io/blog/2023/02/16/d-installer-0-7` blog post URL
+it searches for a discussion with the `blog/2023/02/16/d-installer-0-7` title (or similar, see
+below).
+
+Advantages:
+
+- Giscus can create a new discussion automatically if it does not exist yet
+- Discussion are created only when needed, if nobody comments no discussion is created
+
+Disadvantages:
+
+- The first user who comments needs to authorize the Giscus application, it is not possible to
+  comment at GitHub because the discussion does not exist yet and external users cannot create a new
+  blog post discussion, they can only comment in an existing one
+- Slightly slower loading, Giscus first needs to run a GitHub search query to find the URL of the
+  respective discussion
+- The GitHub search query uses a fuzzy matching, if more blog posts use a similar URL path or title
+  then GitHub might return a wrong discussion as the search result
+
+#### Explicit discussion ID
+
+The respective GitHub discussion ID is defined directly in the blog post front matter. E.g.
+for the `https://github.com/agama-project/agama-project.github.io/discussions/42` discussion URL
+set the `discussion_id: 42` value in the blog post front matter.
+
+Advantages:
+
+- Faster loading, no need to run a GitHub query, the discussion URL is already known
+- Avoids using a wrong discussion because of the fuzzy GitHub search (see above)
+- The first user can add a comment directly to GitHub discussion without need to authorize the
+  Giscus application
+
+Disadvantages:
+
+- The discussion needs to be created manually in advance by the Agama developers, it cannot be
```

Comment:
I wonder if there is no way to automate the discussion creation at the time to merge the PR. Some hook doing the work previously and injecting the id.

That way, the only disadvantage would be to have an empty discussion forever, not a big deal.

Reply:
We would need to create the discussions also for all old blog posts, it's quite difficult to find out what is a "new" or "old" post...